pkg/ddl/tests/partition/multi_domain_test.go

```diff
@@ -1576,5 +1610,562 @@ func TestMultiSchemaTruncatePartitionWithPKGlobal(t *testing.T) {
+				"52 52 52 Original s:3",
+				"53 156 53 Original s:3 Update s:3 f:2",
+				"53 53 53 Original s:3",
+				"54 54 54 InsertODKU s:3 f:2",
+				"54 54 54 Original s:3",
+				"57 160 57 Insert s:3 f:1 Update s:3 f:2",
+				"58 58 58 Insert s:3 f:1",
+				"58 58 58 InsertODKU s:3 f:2",
+				"6 109 6 Insert s:6 f:1 Update s:6 f:2",
+				"61 164 61 Insert s:3 f:2 Update s:3 f:2",
+				"62 62 62 Insert s:3 f:2",
+				"62 62 62 InsertODKU s:3 f:2",
+				"65 168 65 Insert s:3 f:2 Update s:3 f:1",
+				"66 66 66 Insert s:3 f:2",
+				"66 66 66 InsertODKU s:3 f:1",
+				"67 67 67 Insert s:3 f:2",
+				"68 68 68 InsertODKU s:3 f:2",
+				"7 7 7 Insert s:6 f:1",
+				"7 7 7 InsertODKU s:6 f:2",
+				"70 173 70 Original s:2 Update s:2 f:2",
+				"71 71 71 InsertODKU s:2 f:2",
+				"71 71 71 Original s:2",
+				"74 177 74 Insert s:2 f:1 Update s:2 f:2",
+				"75 75 75 Insert s:2 f:1",
+				"75 75 75 InsertODKU s:2 f:2",
+				"78 181 78 Insert s:2 f:2 Update s:2 f:2",
+				"79 79 79 Insert s:2 f:2",
+				"79 79 79 InsertODKU s:2 f:2",
+				"82 185 82 Insert s:2 f:2 Update s:2 f:1",
+				"83 83 83 Insert s:2 f:2",
+				"83 83 83 InsertODKU s:2 f:1",
+				"84 84 84 Insert s:2 f:2",
+				"85 85 85 InsertODKU s:2 f:2",
+				"87 190 87 Original s:1 Update s:1 f:2",
+				"88 88 88 InsertODKU s:1 f:2",
+				"88 88 88 Original s:1",
+				"91 194 91 Insert s:1 f:1 Update s:1 f:2",
+				"92 92 92 Insert s:1 f:1",
+				"92 92 92 InsertODKU s:1 f:2",
+				"95 198 95 Insert s:1 f:2 Update s:1 f:2",
+				"96 96 96 Insert s:1 f:2",
+				"96 96 96 InsertODKU s:1 f:2",
+				"99 202 99 Insert s:1 f:2 Update s:1 f:1"))
+		}
+		logutil.BgLogger().Info("postFn done", zap.Int("state", state))
+	}
+	runMultiSchemaTest(t, createSQL, alterSQL, initFn, postFn, loopFn, true)
+}
+
+func TestIssue58692(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t (a int, b int, index idx(a)) partition by hash(a) partitions 5")
+	tk.MustExec("insert into t (a, b) values (1, 1), (2, 2), (3, 3)")
+	var i atomic.Int32
+	i.Store(3)
+	testfailpoint.EnableCall(t, "github.com/pingcap/tidb/pkg/ddl/afterWaitSchemaSynced", func(job *model.Job) {
+		tk2 := testkit.NewTestKit(t, store)
+		tmp := i.Add(1)
+		tk2.MustExec(fmt.Sprintf("insert into test.t values (%d, %d)", tmp, tmp))
+		tk2.MustExec(fmt.Sprintf("update test.t set b = b + 11, a = b where b = %d", tmp-1))
+	})
+	tk.MustExec("alter table t remove partitioning")
+	rsIndex := tk.MustQuery("select *,_tidb_rowid from t use index(idx)").Sort()
+	rsTable := tk.MustQuery("select *,_tidb_rowid from t use index()").Sort()
+	tk.MustExec("admin check table t")
+	tk.MustQuery("select * from t where b = 20").Check(testkit.Rows("9 20"))
+	tk.MustQuery("select * from t use index(idx) where a = 9").Check(testkit.Rows("9 20"))
+	// TODO: about 1% of the runs, this can fail with '9 20' from table and '9 9' from index
```

Comment:
This may be caused by the fact that after the DDL is completed and return, the SQL in the failpoint is called again(the job state is `JobStateSynced`), and there is a time gap. And I think `TestAddGlobalIndexInIngestWithUpdate` also has this problem.